core/src/test/scala/unit/kafka/server/RequestQuotaTest.scala

```diff
@@ -90,93 +104,142 @@ class RequestQuotaTest extends BaseRequestTest {
     quotaProps.put(QuotaConfigs.PRODUCER_BYTE_RATE_OVERRIDE_CONFIG, "1")
     quotaProps.put(QuotaConfigs.REQUEST_PERCENTAGE_OVERRIDE_CONFIG, "0.01")
-    adminZkClient.changeClientIdConfig(Sanitizer.sanitize(smallQuotaProducerClientId), quotaProps)
+    changeClientIdConfig(Sanitizer.sanitize(smallQuotaProducerClientId), quotaProps)
     quotaProps.put(QuotaConfigs.CONSUMER_BYTE_RATE_OVERRIDE_CONFIG, "1")
     quotaProps.put(QuotaConfigs.REQUEST_PERCENTAGE_OVERRIDE_CONFIG, "0.01")
-    adminZkClient.changeClientIdConfig(Sanitizer.sanitize(smallQuotaConsumerClientId), quotaProps)
+    changeClientIdConfig(Sanitizer.sanitize(smallQuotaConsumerClientId), quotaProps)
 
     TestUtils.retry(20000) {
-      val quotaManager = servers.head.dataPlaneRequestProcessor.quotas.request
+      val quotaManager = brokers.head.dataPlaneRequestProcessor.quotas.request
       assertEquals(Quota.upperBound(0.01), quotaManager.quota("some-user", "some-client"), s"Default request quota not set")
       assertEquals(Quota.upperBound(2000), quotaManager.quota("some-user", unthrottledClientId), s"Request quota override not set")
-      val produceQuotaManager = servers.head.dataPlaneRequestProcessor.quotas.produce
+      val produceQuotaManager = brokers.head.dataPlaneRequestProcessor.quotas.produce
       assertEquals(Quota.upperBound(1), produceQuotaManager.quota("some-user", smallQuotaProducerClientId), s"Produce quota override not set")
-      val consumeQuotaManager = servers.head.dataPlaneRequestProcessor.quotas.fetch
+      val consumeQuotaManager = brokers.head.dataPlaneRequestProcessor.quotas.fetch
       assertEquals(Quota.upperBound(1), consumeQuotaManager.quota("some-user", smallQuotaConsumerClientId), s"Consume quota override not set")
     }
   }
 
+  private def changeClientIdConfig(sanitizedClientId: String, configs: Properties): Unit = {
+    if (isKRaftTest()) {
+      val admin = createAdminClient()
+      admin.alterClientQuotas(Collections.singleton(
+        new ClientQuotaAlteration(
+          new ClientQuotaEntity(Map(ClientQuotaEntity.CLIENT_ID -> (if (sanitizedClientId == "<default>") null else sanitizedClientId)).asJava),
+          configs.asScala.map { case (key, value) => new ClientQuotaAlteration.Op(key, value.toDouble)}.toList.asJava)
+      )).all().get()
+    } else {
+      adminZkClient.changeClientIdConfig(sanitizedClientId, configs)
+    }
+  }
+
   @AfterEach
   override def tearDown(): Unit = {
     try executor.shutdownNow()
     finally super.tearDown()
   }
 
-  @Test
-  def testResponseThrottleTime(): Unit = {
-    for (apiKey <- RequestQuotaTest.ClientActions ++ RequestQuotaTest.ClusterActionsWithThrottle)
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testResponseThrottleTime(quorum: String): Unit = {
+    val apiKeys = if (isKRaftTest()) {
+      (clientActions ++ clusterActionsWithThrottle).filterNot(_.forwardable)
```

Comment:
Currently, `ClientQuotasDelta` is only published to brokers whereas `ControllerApis.QuotaManagers` is not updated after we update client quota configs, so all request is not throttled in the controller, so we need to filter forwardable ApiKeys here.